src/pkg/auditext/dao/dao.go

```diff
@@ -0,0 +1,205 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dao
+
+import (
+	"context"
+	"strings"
+
+	beegorm "github.com/beego/beego/v2/client/orm"
+
+	"github.com/goharbor/harbor/src/common/utils"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/orm"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/auditext/model"
+)
+
+// DAO is the data access object for audit log
+type DAO interface {
+	// Create the audit log ext
+	Create(ctx context.Context, access *model.AuditLogExt) (id int64, err error)
+	// Count returns the total count of audit log ext according to the query
+	Count(ctx context.Context, query *q.Query) (total int64, err error)
+	// List audit log ext according to the query
+	List(ctx context.Context, query *q.Query) (access []*model.AuditLogExt, err error)
+	// Get the audit log ext specified by ID
+	Get(ctx context.Context, id int64) (access *model.AuditLogExt, err error)
+	// Delete the audit log ext specified by ID
+	Delete(ctx context.Context, id int64) (err error)
+	// Purge the audit log ext
+	Purge(ctx context.Context, retentionHour int, includeOperations []string, dryRun bool) (int64, error)
+	// UpdateUsername replaces username in matched records
+	UpdateUsername(ctx context.Context, username string, usernameReplace string) error
+}
+
+// New returns an instance of the default DAO
+func New() DAO {
+	return &dao{}
+}
+
+type dao struct{}
+
+func (d *dao) UpdateUsername(ctx context.Context, username string, usernameReplace string) error {
+	o, err := orm.FromContext(ctx)
+	if err != nil {
+		return err
+	}
+	_, err = o.Raw("UPDATE audit_log_ext SET username = ? WHERE username = ?", usernameReplace, username).Exec()
```

Comment:
Do we need to check the name length as what we did in the `Create` method?

Reply:
Because the new username is autogenerate field, not user input, it is OK to skip the check